discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -725,6 +730,15 @@ def wait_for_cluster_to_be_in_installing_status(self):
             timeout=consts.START_CLUSTER_INSTALLATION_TIMEOUT
         )
 
+    def wait_for_cluster_to_be_in_finalizing_status(self):
+        utils.wait_till_cluster_is_in_status(
+            client=self.api_client,
+            cluster_id=self.id,
+            statuses=[consts.ClusterStatus.FINALIZING],
```

Comment:
Anyway to immediately "timeout" here in case the cluster is in a state from which it can never be finalizing? e.g. error states and "installation complete" state?

Reply:
Oh missed that, cool. Then can you just add `consts.ClusterStatus.INSTALLED` to `statuses=`?

Reply:
I need to get into Finalizing stage and not installed. 
In my e2e case, i wait until Finalizing, because then i can mess up the nodes in order to simulate operator failure & OCP success. 
Then i wait for operators to complete, and then waiting for cluster installed
https://gitlab.cee.redhat.com/ocp-edge-qe/kni-assisted-installer-auto/-/merge_requests/170/diffs

Reply:
> I need to get into Finalizing stage and not installed.
> In my e2e case, i wait until Finalizing, because then i can mess up the nodes in order to simulate operator failure & OCP success.
> Then i wait for operators to complete, and then waiting for cluster installed
> [gitlab.cee.redhat.com/ocp-edge-qe/kni-assisted-installer-auto/-/merge_requests/170/diffs](https://gitlab.cee.redhat.com/ocp-edge-qe/kni-assisted-installer-auto/-/merge_requests/170/diffs)

But usually when someone "Waits for finalizing", then they should also be okay with "INSTALLED". So I suggest you add it to `statuses=` and then in your test, make sure you haven't reached INSTALLED after the call to this function